executor/executor_test.go

```diff
@@ -827,6 +827,17 @@ func (s *testSuite) TestIndexScan(c *C) {
 	tk.MustExec("INSERT INTO tab1 VALUES(1,37,20.85,30,10.69)")
 	result = tk.MustQuery("SELECT pk FROM tab1 WHERE ((col3 <= 6 OR col3 < 29 AND (col0 < 41)) OR col3 > 42) AND col1 >= 96.1 AND col3 = 30 AND col3 > 17 AND (col0 BETWEEN 36 AND 42)")
 	result.Check(testkit.Rows())
+	tk.MustExec("drop table if exists tab1")
```

Comment:
This case can't actually cover you change. Because the "a" may used to calcuate range and result in access conditions.
you should create index (a,b) and involved a column c which is not in any index.